### PR TITLE
#2 - Port MoviesPage

### LINKED ISSUE #2: Port MoviesPage
- fetch movies directly on page

### PR DESCRIPTION
**Pull request by focusedXmind**
## Checklist

Confirm you have completed the following actions prior to submitting this PR.

- [X] There is an existing issue report for this PR.
- [X] I have created a feature branch.
- [X] My changes have been committed.
- [X] I have pushed my changes to the branch.

## Description

Converted MoviesPage to next.js incl. needed components

Closes #2

### CODE REVIEW

File: src/app/(full-page)/moviesPage/page.tsx
Comment (dnmct) [inline src/app/(full-page)/moviesPage/page.tsx:1]: the route name is not correct, should be just movies

File: src/app/(full-page)/moviesPage/page.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            active={currentPage === index + 1}
+            onClick={() => handlePageSelect(index + 1)}
+          />
+        ))}
+      </div>
+    </div>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps = async () => {
```
Comment (dnmct) [inline src/app/(full-page)/moviesPage/page.tsx]: We don't use getServerSideProps with the app router.
Pls use a server component for the page

File: src/app/(full-page)/moviesPage/page.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { MovieCard } from "../../../components/movie-card";
+import { PageButton } from "../../../components/page-button";
+import axios from "axios";
+
+interface MoviesProps {
+  movies: Movie[];
+}
+
+const Movies = ({ movies }: MoviesProps) => {
+  const [currentPage, setCurrentPage] = useState(1);
```
Comment (dnmct) [inline src/app/(full-page)/moviesPage/page.tsx]: You can manage the pages through a query parameter in the url
(e.g. /movies?page=1)
This allows you to keep the page as a server component
The button would then be a links, that navigate to the movies page with the query paramter

**Review by dnmct — CHANGES_REQUESTED** [review]

File: src/app/(full-page)/moviesPage/page.tsx
Comment (focusedXmind) [reply src/app/(full-page)/moviesPage/page.tsx:1]: but there is already a route called "movies", when i try to edit it to movies theres no way to do it. Is it possible to implement the Overview in the existing one?

Comment (dnmct) [reply src/app/(full-page)/moviesPage/page.tsx:1]: Yeah the movies page currently is just a placeholder, you can just put your stuff inside there

Comment (focusedXmind) [reply src/app/(full-page)/moviesPage/page.tsx:1]: aaaahh, easy! :D

File: src/app/(full-page)/movies/[movieId]/page.tsx [hunk 3]
```diff
@@ -1,125 +1,75 @@
...
-      `https://api.themoviedb.org/3/movie/${params.movieId}/credits`,
-      {
-        params: {
-          api_key: process.env.TMDB_API_KEY,
-          language: "en-US",
-        },
-      }
-    )
-    .then((res) => res.data);
+const Movies = ({ movies }: MoviesProps) => {
```
Comment (dnmct) [inline src/app/(full-page)/movies/[movieId]/page.tsx]: Where are you passing the movies from as a prop?
Ususally you cannot pass ciustom props to pages.
Also this is the incorrect page, this page is the movie detail page

**Review by dnmct — CHANGES_REQUESTED** [review]
You still use useState instead of the url for the page state

File: src/app/(full-page)/movies/[movieId]/page.tsx (hunk 3 above)
Comment (focusedXmind) [reply src/app/(full-page)/movies/[movieId]/page.tsx]: sry, uploaded the wrong code, was switching between projects and messed up :D

**Review by dnmct — APPROVED** [review]

**Pull request merged**